vmodules/util/interfaces/BotConfig.ts

```diff
@@ -1,3 +1,15 @@
+import { ColorResolvable, ChannelResolvable, RoleResolvable, PermissionString } from "discord.js";
+import Command from '../../classes/command';
+
+interface VectorPermission {
+  name: string | null,
+  type: "PERM" | "USER" | "ROLE",
+  target: PermissionString, // TODO: check with jack on this
```

Comment:
This too. Is it just a permission string or can it also be a user/role snowflake? in which case it'd be Discord.UserResolvable | Discord.RoleResolvable

Reply:
correct.

`type: "PERM"` means `target` will be a permission flag https://discord.js.org/#/docs/main/stable/class/Permissions?scrollTo=s-FLAGS

`type: "USER"` means `target` will be a user ID

`type: "ROLE"` means `target` will be a role ID